### PR TITLE
[torch] Add OnnxToTorch lowering for `onnx.HannWindow`

### PR DESCRIPTION
**Pull request by vinayakdsci**
Adds OnnxToTorch lowering for the `onnx.HannWindow` op. Also factors out common implementation between the window functions.

### CODE REVIEW

**Comment by vinayakdsci** [comment]
cc @vivekkhandelwal1 @renxida.

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp [hunk 1]
```diff
@@ -35,6 +35,116 @@ static LogicalResult createTorchTransposeOp(ConversionPatternRewriter &rewriter,
...
+
+  Value zero = rewriter.create<Torch::ConstantFloatOp>(
+      binder.getLoc(), rewriter.getF64FloatAttr(0.0));
+  Value one = rewriter.create<Torch::ConstantFloatOp>(
+      binder.getLoc(), rewriter.getF64FloatAttr(1.0));
+  Value two = rewriter.create<Torch::ConstantFloatOp>(
+      binder.getLoc(), rewriter.getF64FloatAttr(2.0));
+
+  constexpr double pi = llvm::numbers::pi;
+  Value tau = rewriter.create<Torch::ConstantFloatOp>(
```
Comment (renxida) [inline lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp:55]: lovely. hello my fellow tau enjoyer

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp [hunk 2]
```diff
@@ -35,6 +35,116 @@ static LogicalResult createTorchTransposeOp(ConversionPatternRewriter &rewriter,
...
+namespace {
+LogicalResult windowFunctionImpl(OpBinder binder,
+                                 ConversionPatternRewriter &rewriter,
+                                 Value size, Value a0, Value a1, Value a2,
+                                 Torch::ValueTensorType resultType,
+                                 int64_t output_datatype, int64_t periodic) {
+
+  double isPeriodicFp = static_cast<double>(periodic);
+
+  Value zero = rewriter.create<Torch::ConstantFloatOp>(
```
Comment (renxida) [inline lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp:47]: consider either using a Loc variable to replace all the binder.getLoc

or use an `mlir::ImplicitLocOpBuilder b(loc, rewriter);` to reduce code complexity & line length. example here: https://github.com/llvm/torch-mlir/pull/2969/files#diff-a89c31dc3f7349d89fe1d51c41ab28d81e2b88530ae79007bb1958e21d864d32R252

in lib/Conversion/TorchOnnxToTorch/OnnxLstmExpander.cpp:249

**Review by renxida — APPROVED** [review]

**Comment by vinayakdsci** [comment]
> lgtm, looks like quite a big chunk of work, good job!
> 
> just to make sure, this has been numerically tested in Shark-TestSuite right?

Yes! I opened a PR implementing the e2e test there.

**Comment by renxida** [comment]
This is tested numerically in https://github.com/nod-ai/SHARK-TestSuite/pull/219

which i just merged

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp (hunk 2 above)
Comment (vinayakdsci) [reply lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp]: Fixed in the latest commit, the code does look much cleaner now. Thanks for this tip!

**Pull request merged**